includes/Core/Modules/Modules.php

```diff
@@ -244,6 +245,41 @@ function ( Module $module ) {
 			}
 		);
 
+		add_action(
+			'googlesitekit_authorize_user',
+			function( $token_response ) {
+				if ( empty( $token_response['analytics_configuration'] ) ) {
+					return;
+				}
+
+				if ( ! $this->is_module_active( Analytics::MODULE_SLUG ) ) {
+					$this->activate_module( Analytics::MODULE_SLUG );
+				}
```

Comment:
I think it would be more accurate to bail early here if the module is already active. All logic that there is below should only run when the module isn't active:
* The call to `handle_token_response_data` would happen already if the module is active, due to the hook in the `Analytics` class.
* The migration of the extra scope is also only relevant in case the module isn't active when receiving the scopes.
```suggestion
				if ( $this->is_module_active( Analytics::MODULE_SLUG ) ) {
					return;
				}
				
				$this->activate_module( Analytics::MODULE_SLUG );
```

Reply:
Updated.